app/src/main/java/com/example/campusguide/search/PointsOfInterest.kt

```diff
@@ -0,0 +1,130 @@
+    var locationListener: SearchLocationListener? = null
+
+    private val nearbyLocations = PlacesApiSearchResultProvider(activity)
+
+    init {
+        if (!Places.isInitialized())
+            Places.initialize(
+                activity.applicationContext,
+                activity.getString(R.string.google_maps_key)
+            )
+
+        placesClient = Places.createClient(activity)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        dialog?.window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+        val view = inflater.inflate(R.layout.points_of_interest_options, container, false)
+
+        view.findViewById<Button>(R.id.cafe)?.setOnClickListener {
+            dismiss();
+            searchPOINearCurrentLocation("Cafe");
+        }
+        view.findViewById<Button>(R.id.restaurant)?.setOnClickListener {
+            dismiss();
+            searchPOINearCurrentLocation("Restaurant");
+        }
+        view.findViewById<Button>(R.id.shopping)?.setOnClickListener {
+            dismiss();
+            searchPOINearCurrentLocation("Shopping mall");
+        }
+        view.findViewById<Button>(R.id.pharmacy)?.setOnClickListener {
+            dismiss();
+            searchPOINearCurrentLocation("Pharmacy");
+        }
+        view.findViewById<Button>(R.id.grocerystore)?.setOnClickListener {
+            dismiss();
+            searchPOINearCurrentLocation("Grocery store");
+        }
+        return view
+    }
+
+    /**
+     * Gets the user's last known most recent location
+     */
+
+    private fun searchPOINearCurrentLocation(poi: String) {
+        return locationProvider.getLocation {
+            GlobalScope.launch {
+                getNearbyPlaces(poi, it)
+            }
+        }
+    }
+
+    private suspend fun getNearbyPlaces(poi: String, location: Location) {
+        activity.runOnUiThread {
+            PopupSearchLocationListener.clearAllMarkers()
+        }
+
+        val places = nearbyLocations.searchNearbyPlaces(poi, location)
+
+        for (place in places.autocompletePredictions) {
+
+            val placeID = place.placeId
+            val placeFields: List<Place.Field> =
+                listOf(Place.Field.ID, Place.Field.NAME, Place.Field.LAT_LNG)
+            val request = FetchPlaceRequest.newInstance(placeID, placeFields)
+
+            placesClient.fetchPlace(request)
+                .addOnSuccessListener { response: FetchPlaceResponse ->
+                    val placeResponse = response.place
+
+                    val searchLocation = SearchLocation(
+                        placeResponse.name!!,
+                        placeResponse.latLng!!.latitude,
+                        placeResponse.latLng!!.longitude,
+                        placeResponse.id!!,
+                        placeResponse.address?:""
+                    )
```

Comment:
Could this be its own function?

Reply:
I unfortunately cannot without having to made another suspend function because this is the response value of the lambda expression